drivers/eks/eks_driver.go

```diff
@@ -241,6 +247,12 @@ func (state *state) validate() error {
 		return fmt.Errorf("virtual network, subnet, and security group must all be set together")
 	}
 
+	if state.AssociateWorkerNodePublicIP != nil &&
+		!*state.AssociateWorkerNodePublicIP &&
+		(state.VirtualNetwork == "" || len(state.Subnets) == 0) {
+		return fmt.Errorf("if private cluster is set a VPC and subnets must also be provided")
```

Comment:
the if checks `or` condition while the error is saying `and` is needed.

Reply:
discussed offline

Reply:
> if private cluster is set
should be if AssociateWorkerNodePublicIP is set to false, VPC and subnets must also be provided